main.cpp

```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
+
+        ElevatorUser();
+        ~ElevatorUser();
+
+        void callElevator();
+        void enterElevator();
+        void setDestination(int newDestinationFloor);
+        void exitElevator();
+        void takeStairsInstead(int newDestinationFloor);
+    };
+};
+
+struct SchoolBag
+{
+    PencilCase schoolPencilCase;
+
+    SchoolBag();
+    ~SchoolBag();
+
+    void packPencilCase();
+    void unpackPencilCase();
+};
+
+struct Building
+{
+    Elevator elevatorA{"A1"};
+    Elevator elevatorB{"B1"};
+    bool isOpen;
+
+    Building();
+    ~Building();
+
+    void closeBuilding();
+    void openBuilding();
+    bool buildingEmpty();
+    void clearBuilding();
+    void callElevatorAAndTakePassenger(int designatedFloor, double passengerWeight);
+};
+
+// ============Pencil Functions ================
+
+PencilCase::Pencil::Pencil() : canColor(false), shade("2HB")
+{	
+    std::cout << "****** Pencil Created ******" << std::endl;
+}
+
+PencilCase::Pencil::~Pencil()
+{
+    std::cout << "****** Pencil Destroyed ******" << std::endl;
+}
+
+void PencilCase::Pencil::sharpen(int rotations)
+{
+    if(!isSharp())
+    {
+        std::cout << "Sharpening..." << std::endl;
+        for(int i = 0; i < rotations; ++i)
+            sharpness += 0.067f;
+
+        if(sharpness > 1.5f && sharpness < 2.5f)
+        {
+            sharpEnough = true;
+            std::cout << "Pencil sharpened" << std::endl;
+        }	
+        else if(sharpness >= 2.5f)
+        {
+            sharpEnough = true;
+            std::cout << "Super sharp!" << std::endl;
+        }
+        else
+        {
+            sharpEnough = false;
+            std::cout << "Needs more sharpening" << std::endl;
+        }	
+    }
+    else
+        std::cout << "Pencil sharp enough" << std::endl;
+}
+
+void PencilCase::Pencil::writeSomething(std::string writing)
+{
+    std::cout << writing << std::endl;
+    std::cout << "Finished writing" << std::endl;
+    sharpness-= 0.089f;
+}
+
+bool PencilCase::Pencil::isSharp()
+{
+    return sharpEnough;
+}
+
+// ===============PencilCase Functions=================
+
+PencilCase::PencilCase() : maxItemsToHold(20), isOpen(false)
+{
+    std::cout << "****** PencilCase Created ******" << std::endl;	
+}
+
+PencilCase::~PencilCase()
+{
+    std::cout << "****** PencilCase Destroyed ******" << std::endl;
+}
+
+void PencilCase::open()
+{
+    isOpen = true;
+}
+
+void PencilCase::close()
+{
+    isOpen = false;
+}
+
+void PencilCase::addItems(int itemsToAdd)
+{
+    if(isOpen)
+    {
+        std::cout << "Adding items to pencilcase..." << std::endl;
+
+        while(numItemsAdded < maxItemsToHold && itemsToAdd > 0)
+        {
+            ++numItemsAdded;
+            --itemsToAdd;
+        }
+        (numItemsAdded == maxItemsToHold) ?  std::cout << "Pencilcase full!" << std::endl : std::cout << numItemsAdded << " items added..." << std::endl;
+    }
+    else
+        std::cout << "Open the pencilcase first" << std::endl;
+}
```

Comment:
If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [ ] revise